frontend/src/features/admin/agencies/EditAgencyPage.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from 'react';
+import { Navbar, Container, Row, ButtonToolbar, Button } from 'react-bootstrap';
+import { Form, Input, Select, SelectOption } from '../../../components/common/form';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+import { IAgencyDetail } from 'interfaces';
+import { Formik, Field } from 'formik';
+import * as API from 'constants/API';
+import './EditAgencyPage.scss';
+import { Label } from 'components/common/Label';
+import { useHistory } from 'react-router-dom';
+import TooltipIcon from 'components/common/TooltipIcon';
+import useCodeLookups from 'hooks/useLookupCodes';
+import { ILookupCode } from 'actions/lookupActions';
+import {
+  createAgency,
+  deleteAgency,
+  fetchAgencyDetail,
+  getUpdateAgencyAction,
+} from 'actionCreators/agencyActionCreator';
+import { FaArrowAltCircleLeft } from 'react-icons/fa';
+import { clear } from 'actions/genericActions';
+import * as actionTypes from 'constants/actionTypes';
+import GenericModal from 'components/common/GenericModal';
+
+interface IEditAgencyPageProps {
+  id: number;
+  match?: any;
+}
+
+const EditAgencyPage = (props: IEditAgencyPageProps) => {
+  const agencyId = props?.match?.params?.id || props.id;
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const newAgency = history.location.pathname.includes('/new');
+  const [showDelete, setShowDelete] = useState(false);
+  useEffect(() => {
+    if (!newAgency) {
+      dispatch(fetchAgencyDetail({ id: agencyId }));
+    }
+  }, [dispatch, agencyId, newAgency]);
+
+  const { getByType } = useCodeLookups();
+  const agencies = getByType(API.AGENCY_CODE_SET_NAME);
+
+  const agency = useSelector<RootState, IAgencyDetail>(
+    state => state.GET_AGENCY_DETAIL as IAgencyDetail,
+  );
+  const mapLookupCode = (code: ILookupCode): SelectOption => ({
+    label: code.name,
+    value: code.id.toString(),
+    selected: !!agency.parentId,
+  });
+  //
+  const selectAgencies = agencies.map(c => mapLookupCode(c));
+
+  const checkAgencies = (
+    <Select
+      label="Parent Agency"
+      field="parentId"
+      data-testid="agency"
```

Comment:
I think I've fixed the selects so that *byLabel works properly now, so data-testid may not be required.

Reply:
Wooo, good news! I'll clean them up